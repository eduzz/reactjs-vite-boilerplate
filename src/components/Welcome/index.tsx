import Box from '@eduzz/houston-ui/Box';
import Button from '@eduzz/houston-ui/Button';
import Typography from '@eduzz/houston-ui/Typography';

type WelcomeProps = {
  to: string;
};

export default function Welcome({ to }: WelcomeProps) {
  return (
    <Box xs={{ padding: 20 }}>
      <Typography size='normal'>Welcome to</Typography>
      <Typography size='xx-large' fontWeight='bold'>
        {to}
      </Typography>
      <p>
        <Button>Click me</Button>
      </p>
    </Box>
  );
}
