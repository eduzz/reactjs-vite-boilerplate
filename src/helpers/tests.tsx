import { FC, ReactElement } from 'react';

import { render, RenderOptions } from '@testing-library/react';

import ThemeProvider from '@eduzz/houston-ui/styles/ThemeProvider';

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
