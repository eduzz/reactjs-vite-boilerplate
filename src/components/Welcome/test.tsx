import Welcome from '@/components/Welcome';
import { render, screen } from '@/helpers/tests';

describe('<Welcome />', () => {
  it('should render the Welcome', () => {
    const { container } = render(<Welcome to='App name' />);

    expect(screen.getByText(/welcome to/i)).toBeInTheDocument();
    expect(screen.getByText(/app name/i)).toBeInTheDocument();
    expect(
      screen.getByRole('button', {
        name: /click me/i
      })
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
