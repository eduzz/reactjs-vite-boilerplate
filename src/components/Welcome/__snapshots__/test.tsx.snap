// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intro /> should render the Intro 1`] = `
<style>
  
        form {
          width: 100%;
        }

        .houston-icon {
          line-height: 0;
        }

        
        @import url('https://fonts.googleapis.com/css2?family=Open+Sans:wght@300;400;600;700');

        body {
          font-family: Open Sans, "Helvetica Neue", Arial, sans-serif;
          font-size: 16px;
        }
      
      
</style>
`;
